test/ZNSStringResolver.test.ts

```diff
@@ -0,0 +1,465 @@
+      await meowToken.mint(operator.address, operatorBalance);
+      await meowToken.connect(operator).approve(await treasury.getAddress(), ethers.MaxUint256);
+
+      userBalance = ethers.parseEther("1000000000000000000");
+      await meowToken.mint(user.address, userBalance);
+      await meowToken.connect(user).approve(await treasury.getAddress(), ethers.MaxUint256);
+
+      deployerBalance = ethers.parseEther("1000000000000000000");
+      await meowToken.mint(deployer.address, deployerBalance);
+      await meowToken.connect(deployer).approve(await treasury.getAddress(), ethers.MaxUint256);
+    });
+
+    afterEach(async () => {
+      await mongoAdapter.dropDB();
+    });
+
+    it("Should not allow non-owner address to setString (similar domain and string)", async () => {
+
+      const curStringDomain = "shouldbrake";
+
+      await registrationWithSetup({
+        zns: campaign.state.contracts,
+        user: operator,
+        domainLabel: curStringDomain,
+        domainContent: ethers.ZeroAddress,
+      });
+
+      await expect(
+        stringResolver.connect(user).setString(hashDomainLabel(curStringDomain), curStringDomain)
+      ).to.be.revertedWith(
+        "ZNSStringResolver: Not authorized for this domain"
+      );
+    });
+
+    it("Should allow OWNER to setString and emit event (similar domain and string)", async () => {
+
+      const curString = "wolf";
+      const hash = hashDomainLabel(curString);
+
+      await registrationWithSetup({
+        zns: campaign.state.contracts,
+        user,
+        domainLabel: curString,
+        domainContent: ethers.ZeroAddress,
+      });
+
+      await expect(
+        stringResolver.connect(user).setString(hash, curString)
+      ).to.emit(
+        stringResolver,
+        "StringSet"
+      ).withArgs(
+        hash,
+        curString
+      );
+
+      expect(
+        await stringResolver.resolveDomainString(hash)
+      ).to.equal(curString);
+    });
+
+    it("Should allow OPERATOR to setString and emit event  (different domain and string)", async () => {
+
+      const curDomain = "wild";
+      const curString = "wildlife";
+      const hash = hashDomainLabel(curDomain);
+
+      await registrationWithSetup({
+        zns: campaign.state.contracts,
+        user: deployer,
+        domainLabel: curDomain,
+        domainContent: ethers.ZeroAddress,
+      });
+
+      await registry.connect(deployer).setOwnersOperator(operator, true);
+
+      await expect(
+        stringResolver.connect(operator).setString(hash, curString)
+      ).to.emit(
+        stringResolver,
+        "StringSet"
+      ).withArgs(
+        hash,
+        curString
+      );
+
+      expect(
+        await stringResolver.resolveDomainString(hash)
+      ).to.equal(curString);
+    });
+
+    it.skip("Should allow REGISTRAR_ROLE to setString and emit event." +
+    "TODO: decide, whether the registrar can set a string  (different domain and string)", async () => {
```

Comment:
don't add "TODOs" to the test name. always add them as comments.
and no, Registrar will not be able to set a string through registration. so you can delete this line